src/app/public/modules/text-editor/toolbar/text-editor-toolbar.component.ts

```diff
@@ -0,0 +1,217 @@
+    this.editorService.execCommand(this.editorId, {
+      command: command,
+      value: value
+    });
+    this.styleState = {
+      ...this.styleState,
+      ...this.editorService.getStyleState(this.editorId)
+    };
+  }
+
+  public toggleFontStyle(currentState: boolean, newState: boolean, command: string): void {
+    if (currentState !== newState) {
+      this.execCommand(command);
+    }
+  }
+
+  public link(): void {
+    const priorSelection = this.editorService.saveSelection(this.editorId);
+    const currentLink = this.editorService.getLink(this.editorId);
+    const inputModal = this.modalService.open(SkyTextEditorUrlModalComponent, [{
+      provide: SkyUrlModalContext,
+      useValue: { urlResult: currentLink }
+    }]);
+    inputModal.closed.subscribe((result: SkyModalCloseArgs) => {
+      if (result.reason === 'save' && priorSelection) {
+        if (!currentLink) {
+          this.editorService.restoreSelection(this.editorId, priorSelection);
+        } else {
+          const anchor = this.editorService.getSelectedAnchorTag(this.editorId);
+          this.editorService.selectElement(this.editorId, anchor);
+        }
+
+        this.execCommand('unlink');
+        if (result.data.target === UrlTarget.None) {
+          // Current window
+          this.execCommand('createLink', result.data.url);
+        } else {
+          // New Window
+          const sText = this.editorService.getCurrentSelection(this.editorId);
+          this.execCommand('insertHTML', '<a href="' + result.data.url + '" rel="noopener noreferrer" target="_blank">' + sText + '</a>');
+        }
+      }
+    });
+  }
+
+  public unlink(): void {
+    let currentSelectionRange = this.editorService.getCurrentSelection(this.editorId).getRangeAt(0);
+    if (currentSelectionRange.toString().length <= 0) {
+      const anchorTag = this.editorService.getSelectedAnchorTag(this.editorId);
+      this.editorService.selectElement(this.editorId, anchorTag);
+    }
+    this.execCommand('unlink');
+  }
+
+  public changeFontSize(size: number): void {
+    this.editorService.setFontSize(this.editorId, size);
+    this.styleState = {
+      ...this.styleState,
+      ...this.editorService.getStyleState(this.editorId)
+    };
+  }
+
+  public onColorpickerColorChanged(color: SkyColorpickerOutput, isBackground = false): void {
+    this.execCommand(isBackground ? 'backColor' : 'foreColor', color.hex);
+  }
+
+  private closeDropdowns(): void {
+    const message: SkyColorpickerMessage = { type: SkyColorpickerMessageType.Close };
+    this.colorpickerStream.next(message);
+    this.backColorpickerStream.next(message);
+    this.fontPickerStream.next({ type: SkyDropdownMessageType.Close });
+    this.fontSizeStream.next({ type: SkyDropdownMessageType.Close });
+  }
+
+  private getFontName(font: string): string {
+    for (let i = 0; i < FONT_LIST_DEFAULTS.length; i++) {
+      if (FONT_LIST_DEFAULTS[i].value === font) {
```

Comment:
Should this be using the default font list or the one supplied to the input?

Reply:
Good catch. This was also masking an additional bug. This has been fixed.